lp-core-platform/lp-cp-xwiki-implementation/src/main/java/eu/learnpad/core/impl/sim/XwikiBridgeInterfaceRestResource.java

```diff
@@ -119,8 +119,28 @@ public void updateConfiguration(String coreFacadeHostname, int coreFacadeHostPor
 	
 	@Override
 	public ProcessData getProcessInfos(String processArtifactId) {
-		// TODO Auto-generated method stub
-		return null;
+		HttpClient httpClient = RestResource.getAnonymousClient();
+		String uri = String.format("%s/learnpad/sim/bridge/processes/%s",
+				RestResource.SIM_REST_URI,processArtifactId);
+
+		GetMethod getMethod = new GetMethod(uri);
+		getMethod.addRequestHeader("Content-Type", "application/json");
+		
+		ProcessData unmashelledProcessData = new ProcessData();
+		
+		try {
+
+			httpClient.executeMethod(getMethod);
+		    
+			ObjectMapper objectMapper = new ObjectMapper();
+// Not fully tested, but is looks working for our purposes -- Gulyx
+			unmashelledProcessData = objectMapper.readValue(getMethod.getResponseBodyAsStream(), ProcessData.class);			
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		
+		return unmashelledProcessData;
```

Comment:
The returned object may not have been correctly initialized in case of an IOException during try-block on line 131. In this case it will return the dummy ProcessData object created on line 129.

With this configuration, callers of getProcessInfos must be aware that the method may return a dummy object in case of failure. I think it may be somewhat dangerous as it may cause problems difficult to debug.

The "proper" solution would be to modify the interface to declare that all methods can throw an Exception. I see that this is the way that have been adopted in the [mv component](https://github.com/gulyx/learnpad/blob/cp-SIM-BridgeRestResource/lp-core-platform/lp-cp-apis/src/main/java/eu/learnpad/mv/rest/Verification.java). We should probably modify sim interfaces to do the same.

I will make the change today, so we can do it cleanly.